geometry.zsh

```diff
@@ -270,6 +278,35 @@ prompt_geometry_clear_timestamp() {
   unset prompt_geometry_command_exec_time
 }
 
+GEOMETRY_ASYNC_PROMPT_PROC=0
+prompt_geometry_setup_async_prompt() {
+    function precmd() {
```

Comment:
`precmd` runs before each prompt, but does it interfere with other `precmd` hook functions?